dlib/test/geometry.cpp

```diff
@@ -895,6 +895,39 @@ namespace
 
     }
 
+// ----------------------------------------------------------------------------------------
+
+    void test_polygon()
```

Comment:
I've added a couple of checks to test the area and make sure that the `get_convex_shape` does what we expect.

Let me know if that is enough.

Reply:
Yeah, nice.